### PR TITLE
Support building Red Hat 7 AMIs

### PR DESCRIPTION
**Pull request by sschneid**
Support for building on Red Hat Linux

* new `rhel7/` folder/image and build scripts
* added `lib` folder with shared provisioning scripts
* reworked `centos7/` scripts to use shared provisioning scripts
* updated `README.md` to be more platform-agnostic

### CODE REVIEW

**Comment by sschneid** [comment]
cc @lingmann @margaret

**Comment by sschneid** [comment]
@mesosphere-mergebot bump-ee

**Comment by sschneid** [comment]
@mesosphere-mergebot bump-ee

File: cloud_images/README.md [hunk 1]
```diff
@@ -1,27 +1,45 @@
...
-Steps to create a new AMI with the DC/OS pre-requisites installed using a base CentOS 7 AMI (Marketplace or otherwise) and [Packer](https://www.packer.io/).
+2. Copy the `centos7` folder to `${HOST}`, then connect via SSH:
+```
+$ scp -r centos7 centos@${HOST}:.
+$ ssh centos@${HOST}
+```
+
+The following steps are performed on the launched AMI:
+
+3. `cd` into th` centos7` folder and run `create_base_ami.sh` specifying the secondary EBS volume's raw device:
```
Comment (colin-msphere) [inline cloud_images/README.md]: spelling: `into the centos7 folder`

File: cloud_images/README.md [hunk 2]
```diff
@@ -1,27 +1,45 @@
...
+6. Create AMI from snapshot
+
+   Name: `centos7-YYYYMMDDhhmm` (use same value as snapshot),
+   Virtualization type: `Hardware-assisted virtualization`
+7. Update AMI Permissions to 'Public'
+8. Record in `CHANGELOG.md` new AMI details
+
+## Create DC/OS-ready AMI
+
+The following steps will create a new AMI with the DC/OS pre-requisites installed a base AMI and [Packer](https://www.packer.io/):
```
Comment (colin-msphere) [inline cloud_images/README.md]: not sure: `installed a base AMI`

File: cloud_images/centos7/create_base_ami.sh [hunk 3]
```diff
@@ -10,159 +10,39 @@ set -o errexit -o nounset -o pipefail
 
 : ${DEVICE:?"ERROR: DEVICE must be set"}
 
-ROOTFS=/rootfs
-PARTITION=${DEVICE}1
+OS_VERSION=7-4.1708.el7.centos
+OS_VERSION_SHORT=7.4.1708
```
Comment (colin-msphere) [inline cloud_images/centos7/create_base_ami.sh]: Previously this was configurable: `export CENTOS_VERSION=${CENTOS_VERSION:-"7.4.1708"}` - should it still be here?

Comment (sschneid) [reply cloud_images/centos7/create_base_ami.sh]: The version variables is only used by `create_base_ami.sh` to install the `centos-release` package into the rootfs.  It was previously exported and used to `yum update` the server; `yum update` should not be necessary, as the images should be built from the latest (`7.4` in this case) AMI from the AWS Marketplace (step #1).

**Comment by sschneid** [comment]
@mesosphere-mergebot bump-ee

**Comment by sschneid** [comment]
@mesosphere-mergebot label Ready For Review

**Comment by sschneid** [comment]
@mesosphere-mergebot bump-ee

**Pull request closed without merging**